lib/msf/core/auxiliary/report.rb

```diff
@@ -291,10 +291,9 @@ def report_vuln(opts={})
         :task => mytask
     }.merge(opts)
     vuln = framework.db.report_vuln(opts)
-    if vuln.nil?
-      print_error("Failed to report vuln for #{opts[:host]}:#{opts[:port]} to the database")
-      return
-    end
+
+    raise AuxiliaryError, "Failed to report vuln for #{opts[:host]}:#{opts[:port]} to the database" if vuln.nil?
```

Comment:
It may make more sense to use `ValidationError` here, see https://github.com/rapid7/metasploit-framework/blob/04e8752b9b74cbaad7cb0ea6129c90e3172580a2/lib/msf/core/payload/windows/pe_inject.rb#L39-L40

As per https://github.com/rapid7/metasploit-framework/blob/588e91ae276851eb8b3b123f5bf78aae4f3132be/lib/msf/core/exception.rb#L47-L58

we can see this error is meant to be used when something doesn't pass validation which will be the case here. Initially I was hesitant on this but this is operating on the `opts` parameter so the fact that this inherits from `ArgumentError` should make sense in this circumstance.

Reply:
This makes sense, and it is much more specific than the generic AuxiliaryError. Fixed in commit `c6a84c9`